### PR TITLE
Fix pagination in SCIM2 listUsers

### PR DESCRIPTION
**Pull request by IshikaGodage**
Fix [wso2/product-is#4601](https://github.com/wso2/product-is/issues/4601)
Fix [wso2/product-is#4604](https://github.com/wso2/product-is/issues/4604)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -282,7 +282,7 @@
         <cxf-bundle.version>3.2.8</cxf-bundle.version>
         <inbound.auth.oauth.version>5.2.0</inbound.auth.oauth.version>
         <commons-collections.version>3.2.0.wso2v1</commons-collections.version>
-        <carbon.kernel.version>4.4.34</carbon.kernel.version>
+        <carbon.kernel.version>4.4.39-SNAPSHOT</carbon.kernel.version>
```
Comment (DMHP) [inline pom.xml]: Change this to 4.4.39

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 2]
```diff
@@ -1119,15 +1204,18 @@ private Condition createConditionForSingleAttributeFilter(String domainName, Nod
...
-
+        if (StringUtils.isEmpty(domainName)) {
+            domainName = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
+        }
         try {
-            givenMax = Integer.parseInt(carbonUM.getRealmConfiguration().getUserStoreProperty(
-                    "MaxUserNameListLength"));
+            givenMax = Integer.parseInt(carbonUM.getSecondaryUserStoreManager(domainName).getRealmConfiguration()
+                    .getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_MAX_USER_LIST));
         } catch (Exception e) {
```
Comment (DMHP) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: Can we use a generic exception rather catching the Exception

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 3]
```diff
@@ -1119,15 +1204,18 @@ private Condition createConditionForSingleAttributeFilter(String domainName, Nod
...
 
         int givenMax;
-
+        if (StringUtils.isEmpty(domainName)) {
+            domainName = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
+        }
         try {
-            givenMax = Integer.parseInt(carbonUM.getRealmConfiguration().getUserStoreProperty(
-                    "MaxUserNameListLength"));
+            givenMax = Integer.parseInt(carbonUM.getSecondaryUserStoreManager(domainName).getRealmConfiguration()
```
Comment (DMHP) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: Can carbonUM.getSecondaryUserStoreManager(domainName).getRealmConfiguration().getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_MAX_USER_LIST) be null? Can we check null and assign a default value if it is null.

File: pom.xml (hunk 1 above)
Comment (IshikaGodage) [reply pom.xml]: Fixed

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java (hunk 2 above)
Comment (IshikaGodage) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: Fixed

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java (hunk 3 above)
Comment (IshikaGodage) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: Fixed

**Pull request merged**